webapp/components/Editor/Editor.vue

```diff
@@ -200,12 +200,173 @@ export default {
     EditorMenuBubble,
   },
   props: {
-    users: { type: Array, default: () => [] },
-    hashtags: { type: Array, default: () => [] },
+    users: { type: Array, default: () => null }, // If 'null', than the Mention extention is not assigned.
+    hashtags: { type: Array, default: () => null }, // If 'null', than the Hashtag extention is not assigned.
     value: { type: String, default: '' },
     doc: { type: Object, default: () => {} },
   },
   data() {
+    // Set array of optional extensions by analysing the props.
+    let optionalExtensions = []
+    // Don't change the following line. The functionallity is in danger!
+    if (this.users) {
+      optionalExtensions.push(
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-08-19T14:37:28Z" title="Monday, August 19th 2019, 4:37:28 pm +02:00">Aug 19, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -200,12 +200,173 @@ export default {
     EditorMenuBubble,
   },
   props: {
-    users: { type: Array, default: () => [] },
-    hashtags: { type: Array, default: () => [] },
+    users: { type: Array, default: () => null }, // If 'null', than the Mention extention is not assigned.
+    hashtags: { type: Array, default: () => null }, // If 'null', than the Hashtag extention is not assigned.
     value: { type: String, default: '' },
     doc: { type: Object, default: () => {} },
   },
   data() {
+    // Set array of optional extensions by analysing the props.
+    let optionalExtensions = []
+    // Don't change the following line. The functionallity is in danger!
+    if (this.users) {
+      optionalExtensions.push(
```


Wow, this is a monster component! 👹

I'm not sure I get everything that's going on here but it seems to me that the `Mention` props are pretty similar to the `Hashtag` props so I'm wondering if it would be possible to define all that `onEnter`, `onExit`, `onKeyDown` stuff once and then reuse it to create both the `Mention` and `Hashtag` extension?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-20T05:40:25Z" title="Tuesday, August 20th 2019, 7:40:25 am +02:00">Aug 20, 2019</time>_

---


But we would need still the possibility of this options … @alina-beck 

I would love if you show me how to get this in one class !!! 🤗😍

But later, I suggest, as a refactoring. Just to get this merged and start testing as soon as possible.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-20T06:12:11Z" title="Tuesday, August 20th 2019, 8:12:11 am +02:00">Aug 20, 2019</time>_

---


I agree, we can refactor later, if that's ok